views/js/runner/plugins/tools/highlighter/plugin.js

```diff
@@ -168,35 +274,70 @@ define([
-                                    }
-                                })
-                                .then(function(){
-                                    //save highlighter state during the item session,
-                                    //when the highlighting ends
-                                    highlighter.on('end.save', function(){
-                                        return save();
+
+                            collection.getItemHighlighter().enable();
+                            // Load item-level highlights from store:
+                            loadHighlight(itemId);
+
+                            // Count stimuli in this item:
+                            textStimuli = itemHelper.getTextStimuliHrefs(testRunner);
+
+                            // NOW we can instantiate the extra highlighters:
+                            _.forEach(textStimuli, function(textStimulusHref) {
+                                var stimHighlighter = collection.getHighlighterById(textStimulusHref);
+                                // Instantiate, if id not already present in highlighters...
+                                if (!stimHighlighter) {
+                                    stimHighlighter = collection.addHighlighter({
+                                        className: 'txt-user-highlight',
+                                        containerSelector: '.qti-include[data-href="' + textStimulusHref + '"]',
+                                        id: textStimulusHref
                                     });
-                                });
+                                }
+                                stimHighlighter.enable();
+                                // And load its data:
+                                loadHighlight(null, textStimulusHref);
+                            });
                         }
                     })
+                    .after('renderitem', function() {
+                        // Attach start/end listeners to all highlighter instances:
+                        _.forEach(collection.getAllHighlighters(), function(instance) {
+                            if (instance.isEnabled()) {
+                                instance
+                                    .on('start', function(){
+                                        self.buttonMain.turnOn();
+                                        self.trigger('start');
+                                        hasHighlights = true;
+                                    })
+                                    .on('end', function(){
+                                        self.buttonMain.turnOff();
+                                        self.trigger('end');
+                                    });
+                            }
+                        });
+                    })
+                    .after('clear.highlighter', function() {
+                        collection.saveAll();
```

Comment:
It looks like you should call `saveAll` instead of `collection.saveAll`

Reply:
Thanks for the catch!